telegram/_update.py

```diff
@@ -444,6 +446,56 @@ def effective_user(self) -> Optional["User"]:
         self._effective_user = user
         return user
 
+    @property
+    def effective_sender(self) -> Optional[Union["User", "Chat"]]:
+        """
+        :class:`telegram.User` or :class:`telegram.Chat`: The user or chat that sent this update,
+        no matter what kind of update this is.
+
+        Note:
+            * When the user is anonymous this is either :class:`telegram.Chat` or :obj:`None`.
+            * If the user is not anonymous it's either :class:`telegram.User` or :obj:`None`.
+
+        If no user whatsoever is associated with this update, this gives :obj:`None`. This
+        is the case if any of
+
+        * :attr:`channel_post`
+        * :attr:`edited_channel_post`
```

Comment:
I strongly suggest using `message.sender_chat` also in these cases. Not including a field called `sender_chat` in this property doesn't make much sense to me :D